File: Easy/sessione/meta_sessionedefaultview/meta_sessionedefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_sessionedefaultview
{
    public class Meta_sessionedefaultview : Meta_easydata 
	{
        public Meta_sessionedefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "sessionedefaultview") {
				Name = "Sessioni di esami";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idsessione"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "appellokind_title", "Tipologia di appello", nPos++);
						DescribeAColumn(T, "sessionekind_title", "Tipologia", nPos++);
						DescribeAColumn(T, "sessione_start", "Data di inizio", nPos++);
						DescribeAColumn(T, "sessione_stop", "Data di fine", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/sessione/meta_sessionedefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_sessionedefaultview() {
        MetaData.apply(this, ["sessionedefaultview"]);
        this.name = 'meta_sessionedefaultview';
    }

    meta_sessionedefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_sessionedefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'start', 'Data di inizio', null, 1000, null);
						this.describeAColumn(table, 'sessione_stop', 'Data di fine', null, 2000, null);
						this.describeAColumn(table, 'sessionekind_title', 'Tipologia', null, 3200, 50);
						this.describeAColumn(table, 'appellokind_title', 'Tipologia di appello', null, 4200, 50);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idsessione"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "start desc, sessionekind_title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('sessionedefaultview', new meta_sessionedefaultview('sessionedefaultview'));

	}());
